### PR TITLE
Add `wcadmin_pfw_domain_verify_success` event

### PR DESCRIPTION
**Pull request by tomalec**
### Changes proposed in this Pull Request:

- Fix `pfw_domain_verify_failure` tests, 
   Mock ESM default export, make the mock sync.
- Fire `wcadmin_pfw_domain_verify_success` event, 
   implements part of #232

### Screenshots:

verify

### Detailed test instructions:

0. Start ngrok, or other tunnel to publish your local site.
1. Enable track events logging, with `localStorage.setItem( 'debug', 'wc-admin:*' );`
2. Disconnect your account if needed `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fconnection`
1. Start the onboarding `/wp-admin/admin.php?page=wc-admin&path=%2Fpinterest%2Fonboarding`
2. Follow the wizard, to step 2
3. Click "Start verification"
4. Once successfully verified, you should see `wcadmin_pfw_domain_verify_success` fired.

### CODE REVIEW

**Comment by tomalec** [comment]
After this PR is merge the feature branch PR should be ready for the review https://github.com/woocommerce/pinterest-for-woocommerce/pull/296

File: assets/source/setup-guide/app/steps/ClaimWebsite.test.js [hunk 1]
```diff
@@ -48,4 +57,25 @@ describe( 'Claim Website Record Events', () => {
...
+		fireEvent.click( getByText( 'Start verification' ) );
+
+		// Wait for async click handler and apiFetch resolution.
+		const delay = sleep( 1 );
+		jest.runAllTimers();
+		await delay;
+
+		expect( recordEvent ).toHaveBeenCalledWith(
+			'pfw_domain_verify_success'
+		);
```
Comment (puntope) [inline assets/source/setup-guide/app/steps/ClaimWebsite.test.js]: 💅 

WDYT about...

```
await waitFor( () =>
			expect( recordEvent ).toHaveBeenCalledWith(
				'pfw_domain_verify_success'
			)
		);
```

more info [here](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor)

File: assets/source/setup-guide/app/steps/ClaimWebsite.test.js [hunk 2]
```diff
@@ -32,9 +33,17 @@ import { fireEvent, render } from '@testing-library/react';
  */
 import ClaimWebsite from './ClaimWebsite';
```
Comment (puntope) [inline assets/source/setup-guide/app/steps/ClaimWebsite.test.js:28]: 💅  

Could we remove as well

`jest.mock( '@woocommerce/tracks'`

I guess I forgot myself to remove it when we implemented this in `test/dependencies` as discussed.

**Review by puntope — COMMENTED** [review]
✅  LGTM

✅  I tested the domain verification and worked for me :)
✅  Visual code verification looks good 

💅  Some opinionated comments regarding the tests

File: assets/source/setup-guide/app/steps/ClaimWebsite.test.js (hunk 1 above)
Comment (tomalec) [reply assets/source/setup-guide/app/steps/ClaimWebsite.test.js]: I like it :)
`setTimout(1)` was testing that it's fired immediately after, but `waitFor` should be less fragile` is more readable, and shorter.

Done in https://github.com/woocommerce/pinterest-for-woocommerce/pull/314/commits/de3d9e7646373829b3b942adef0626afc8b06098

We could think of something more like
```js
await waitForNetworkIdle(); // await waifForApiFetchMock()
expect( //...
```
to make sure we fire it synchronously after  API fetch response. But let's not overdo it.

**Comment by tomalec** [comment]
Sorry for the force pushes, I struggled with accidentally staged `.pot` file

**Comment by tomalec** [comment]
Thanks for the review @puntope I pushed both of your suggestions. Would you mind another look?

**Review by puntope — APPROVED** [review]

**Pull request merged**